scala/private/phases/phase_dependency.bzl

```diff
@@ -55,6 +56,11 @@ def _phase_dependency(
     else:
         strict_deps_mode = get_strict_deps_mode(ctx)
 
+    if strict_deps_always_off or not included_in_strict_deps_analysis:
```

Comment:
I left it this way because I think we will need a separate flag to control compiler deps mode in the future. Because it's a breaking change and I'm not very sure about the need yet, I left it this way.

I think I'll squash strict and compiler modes under one control statement. Eg.:
```starlark
    if strict_deps_always_off or not included_in_strict_deps_analysis:
        strict_deps_mode = "off"
        compiler_deps_mode = "off"
    else:
        strict_deps_mode = get_strict_deps_mode(ctx)
        compiler_deps_mode = get_compiler_deps_mode(ctx)
```